src/main/kotlin/com/chattriggers/ctjs/api/world/TabList.kt

```diff
@@ -1,73 +1,66 @@
 package com.chattriggers.ctjs.api.world
 
 import com.chattriggers.ctjs.MCTeam
+import com.chattriggers.ctjs.api.CTWrapper
 import com.chattriggers.ctjs.api.client.Client
 import com.chattriggers.ctjs.api.client.Player
+import com.chattriggers.ctjs.api.entity.Team
 import com.chattriggers.ctjs.api.message.TextComponent
+import com.chattriggers.ctjs.internal.mixins.ClientPlayNetworkHandlerAccessor
+import com.chattriggers.ctjs.internal.mixins.MinecraftClientAccessor
+import com.chattriggers.ctjs.internal.mixins.PlayerListEntryAccessor
 import com.chattriggers.ctjs.internal.mixins.PlayerListHudAccessor
 import com.chattriggers.ctjs.internal.utils.asMixin
 import com.google.common.collect.ComparisonChain
 import com.google.common.collect.Ordering
+import com.mojang.authlib.GameProfile
+import gg.essential.elementa.state.BasicState
 import net.minecraft.client.network.PlayerListEntry
 import net.minecraft.scoreboard.ScoreboardDisplaySlot
+import net.minecraft.scoreboard.ScoreboardObjective
 import net.minecraft.text.Text
+import net.minecraft.util.ApiServices
 import net.minecraft.world.GameMode
+import java.util.*
 
 object TabList {
+    private var needsUpdate = true
+    private var tabListNames = mutableListOf<Name>()
     private val playerComparator = Ordering.from(PlayerComparator())
+    internal var customHeader = false
+    internal var customFooter = false
+    private var tabListHeader = TextComponent("")
+    private var tabListFooter = TextComponent("")
```

Comment:
Is there a reason you chose not to use null for these?

Reply:
No, will try that out now.